cmd/medius/images/common.go

```diff
@@ -0,0 +1,72 @@
+package images
+
+import (
+	"fmt"
+	"path"
+	"sync"
+	"time"
+
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+)
+
+func spawnWorkers(workers int, focus string, workerFn func(api.Artifact) error) (*sync.WaitGroup, chan error) {
+	count := len(common.Registry)
+	errChan := make(chan error, count)
+	jobChan := make(chan api.Artifact, count)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(workers)
+	for x := 0; x < workers; x++ {
```

Comment:
Maybe it would make sense to limit the number of workers, so it is not greater than the number of jobs.
The worker count is a user parameter, so it could be large.

Reply:
Resolved